arangod/RestServer/UpgradeFeature.cpp

```diff
@@ -339,4 +372,28 @@ void UpgradeFeature::upgradeLocalDatabase() {
       << "finished database init/upgrade";
 }
 
+Result UpgradeFeature::performFullCompaction() {
+  LOG_TOPIC("e8f45", INFO, arangodb::Logger::ENGINES)
+      << "starting full RocksDB compaction after upgrade";
+
+  TRI_ASSERT(server().hasFeature<EngineSelectorFeature>());
+  StorageEngine& engine = server().getFeature<EngineSelectorFeature>().engine();
+
+  // Perform full compaction with both changeLevel and compactBottomMostLevel
+  // enabled This matches the behavior of the /_admin/compact API with
+  // bottomMost=true and changeLevels=true
+  Result res = engine.compactAll(true, true);
+
+  if (res.fail()) {
+    LOG_TOPIC("e8f46", FATAL, arangodb::Logger::ENGINES)
+        << "full RocksDB compaction after upgrade failed: "
+        << res.errorMessage();
```

Comment:
I suggest to print `res.errorNumber()` as well, just to be sure we don't miss possibly helpful information.

Reply:
Done.